### ISSUE TITLE
Product of ones is zero on cuda

### ISSUE THREAD

**Issue by justheuristic** [pytorch/pytorch#20019]
## 🐛 Bug

Product of ones is zero on cuda. This seems like a rather common issue, but i found no mentions of it here after a brief keyword search.

## To reproduce
```
import torch
print(torch.__version__)
print(torch.prod(torch.ones(10 ** 6, device='cuda')))
```
prints
```
1.0.1.post2
tensor(0., device='cuda:0')
```

## Expected behavior
I'd expect it to be 1 :)

## Environment
```
Collecting environment information...
PyTorch version: 1.0.1.post2
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.1 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.2) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: 
GPU 0: Tesla M40
GPU 1: Tesla M40
GPU 2: Tesla M40
GPU 3: Tesla M40

Nvidia driver version: 396.37
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.6.0.21
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.1.4

Versions of relevant libraries:
[pip] numpy==1.16.3
[pip] numpydoc==0.7.0
[pip] torch==1.0.1.post2
[pip] torchvision==0.2.1
[conda] blas                      1.0                         mkl  
[conda] faiss-gpu                 1.4.0           py36_cuda8.0.61_1    pytorch
[conda] mkl                       2018.0.1             h19d6760_4  
[conda] mkl-service               1.1.2            py36h17a0993_4  
[conda] pytorch                   1.0.1           py3.6_cuda8.0.61_cudnn7.1.2_2    pytorch
[conda] torch                     1.0.1.post2               <pip>
[conda] torchvision               0.2.1                      py_2    pytorch
```

## Additional context
This does not seem to be a floating point precision issue:
```
print(torch.prod(torch.ones(130_560, device='cuda'), dim=0))
print(torch.prod(torch.ones(130_561, device='cuda'), dim=0))
```
prints
```
tensor(1., device='cuda:0')
tensor(0., device='cuda:0')
```

**Comment by vfdev-5** [comment]
@justheuristic this is fixed in 1.1.0 version (probably by https://github.com/pytorch/pytorch/pull/15653)
Your issue possibly a duplicate of https://github.com/pytorch/pytorch/issues/15647

**Comment by justheuristic** [comment]
Thanks / sorry for inconvenience
